pkg/operator/ceph/object/user/controller.go

```diff
@@ -296,6 +296,20 @@ func (r *ReconcileObjectStoreUser) createOrUpdateCephUser(u *cephv1.CephObjectSt
 	user, err = r.objContext.AdminOpsClient.GetUser(r.opManagerContext, *r.userConfig)
 	if err != nil {
 		if errors.Is(err, admin.ErrNoSuchUser) {
+			namspacedName := types.NamespacedName{Namespace: u.Namespace, Name: "rook-ceph-object-user-" + u.Spec.Store + u.Name}
+			cephObjectStoreUserSecret := &corev1.Secret{}
+			err := r.client.Get(r.clusterInfo.Context, namspacedName, cephObjectStoreUserSecret)
+			if err != nil {
+				if kerrors.IsNotFound(err) {
+					logger.Debugf("no user secret %q provided for cephobjectuser, using random values", "rook-ceph-object-user-"+u.Spec.Store+u.Name)
+				} else {
+					return errors.Wrapf(err, "failed to get user cephobjectuser secret %q", "rook-ceph-object-user-"+u.Spec.Store+u.Name)
+				}
+			} else {
+				r.userConfig.Keys[0].AccessKey = string(cephObjectStoreUserSecret.Data["AccessKey"])
+				r.userConfig.Keys[0].SecretKey = string(cephObjectStoreUserSecret.Data["SecretKey"])
+			}
+
 			user, err = r.objContext.AdminOpsClient.CreateUser(r.opManagerContext, *r.userConfig)
```

Comment:
in the original ask it was said if the secret is updated the cephobjectstoreuser should also get update, so that means do we need to watch for this secret in the cephobjectstoreuser reconcile?

Reply:
or maybe I can simply call with modify user?

Reply:
And we already check for non crd objects